src/main.rs

```diff
@@ -39,45 +43,36 @@ fn main() {
 
     // === Sign in, if user has a token ===
     if !(*auth_store).borrow().is_empty() {
-        fetch_data(&ui, &auth_store.clone());
+        let mut auth_store = (*auth_store).borrow_mut();
+        let mut auths_to_remove = vec![];
+        for (i, auth) in auth_store.iter_mut().enumerate() {
+            match fetch_data(auth.platform.clone(), auth.token.clone(), &ui) {
+                Ok(..) => (),
+                (Err(error)) if error.status() == StatusCode::Unauthorized => {
+                    eprintln!("Token expired");
+                    auths_to_remove.push(i);
+                }
+                _ => eprintln!("There has been an issue with internet connection"),
```

Comment:
This has to be more generic, as there are more errors then just a connection issue that might result in this, I belive.